src/hardware/serialport/serialport.cpp

```diff
@@ -96,43 +96,22 @@ device_COM::~device_COM() {
 CSerial *serialports[SERIAL_MAX_PORTS] = {0};
 
 static Bitu SERIAL_Read (Bitu port, Bitu iolen) {
-	Bitu i;
-	Bitu retval;
-	Bitu index = port & 0x7;
-	switch(port&0xff8) {
-		case 0x3f8: i=0; break;
-		case 0x2f8: i=1; break;
-		case 0x3e8: i=2; break;
-		case 0x2e8: i=3; break;
```

Comment:
This logic seems to be removed by mistake?

Reply:
Darn, this might have been dragged in with `add -p` either by accident or because the changes were too close together and couldn't split further; reformatted chunk should show up in a later commit. Will track it down.

Reply:
Found it; it was an add -p issue, because this block follows directly under a change that was put into a later commit. 

https://github.com/dosbox-staging/dosbox-staging/commit/2e3729deac2c9cde3483a3ad49bc2434456f5abc

If I could, I would like to bring just this changed block back into this commit to make sure the commits can be bisected.  

Is that possible without too much git surgery?